mavis/ExtensionManager.hpp

```diff
@@ -1685,8 +1917,65 @@ namespace mavis::extension_manager
                            SharedPtrAllocator<AnnotationType>()) const
         {
             return constructMavis_<InstType, AnnotationType, InstTypeAllocator,
-                                   AnnotationTypeAllocator>(anno_files, inst_allocator,
-                                                            annotation_allocator);
+                                   AnnotationTypeAllocator>(anno_files, {}, {}, {}, {},
+                                                            inst_allocator, annotation_allocator);
+        }
+
+        // Gets the name that the extension manager would use for a Mavis context based on the
+        // currently enabled extensions
+        const std::string & getContextName() const
+        {
+            return enabled_extensions_.getSortedExtensions();
+        }
+
+        // Switches a Mavis object's context to the currently enabled extensions.
+        // Manages context names internally and should only be used with a Mavis object created with
+        // ExtensionManager::constructMavis
+        template <typename InstType, typename AnnotationType, typename InstTypeAllocator,
+                  typename AnnotationTypeAllocator>
+        void switchMavisContext(
+            Mavis<InstType, AnnotationType, InstTypeAllocator, AnnotationTypeAllocator> & mavis) const
+        {
+            static const std::string BASE_CONTEXT{"BASE"};
```

Comment:
@kathlenemagnus you can use this method to seamlessly switch Mavis contexts as you enable/disable extensions in the manager.